src/libostree/ostree-fetcher-curl.c

```diff
@@ -330,7 +331,14 @@ check_multi_info (OstreeFetcher *fetcher)
             }
           else
             {
-              g_task_return_new_error (task, G_IO_ERROR, G_IO_ERROR_FAILED,
+              /* When it is not a file and the OSTREE_CURL_IO_RETRY
+               * environment variable is set, we want to retry the request.
+               * We accomplish that by using G_IO_ERROR_TIMED_OUT.
+               */
+              should_retry = g_getenv ("OSTREE_CURL_IO_RETRY") != NULL;
```

Comment:
So in GLib there is this handy API `g_parse_debug_string` which is intended for things like this and we use it e.g. over here https://github.com/ostreedev/ostree/blob/92d25eb56bcb17ee117ef5b877de86dc828e823f/src/libostree/ostree-sysroot.c#L192

The advantage of this is it provides a more uniform way to set multiple options/flags from a single environment variable; the UX would be e.g. `env OSTREE_FETCHER_FLAGS=retry-all` or so.

And then let's say later we want to add *another* flag (actually, supporting disabling http2 e.g. might be a good one) we can add `OSTREE_FETCHER_FLAGS=no-http2,retry-all`.